internal/controller/standalone_pgadmin/users.go

```diff
@@ -0,0 +1,286 @@
+)
+
+type Executor func(
+	ctx context.Context, stdin io.Reader, stdout, stderr io.Writer, command ...string,
+) error
+
+type secretPGAdminUser struct {
+	// Whether the user has admin privileges or not.
+	IsAdmin bool `json:"isAdmin"`
+
+	// The user's password
+	Password string `json:"password"`
+
+	// The username for User in pgAdmin.
+	// Must be unique in the pgAdmin's users list.
+	Username string `json:"username"`
+}
+
+// reconcilePGAdminUsers reconciles the default admin user and the users listed in the pgAdmin spec,
+// adding them to the pgAdmin secret, and creating/updating them in pgAdmin when appropriate.
+func (r *PGAdminReconciler) reconcilePGAdminUsers(ctx context.Context, pgadmin *v1beta1.PGAdmin) error {
+	const container = naming.ContainerPGAdmin
+	var podExecutor Executor
+	log := logging.FromContext(ctx)
+
+	// Find the running pgAdmin container. When there is none, return early.
+	pod := &corev1.Pod{ObjectMeta: naming.StandalonePGAdmin(pgadmin)}
+	pod.Name += "-0"
+
+	err := errors.WithStack(r.Client.Get(ctx, client.ObjectKeyFromObject(pod), pod))
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	var running bool
+	for _, status := range pod.Status.ContainerStatuses {
+		if status.Name == container {
+			running = status.State.Running != nil
+		}
+	}
+	if terminating := pod.DeletionTimestamp != nil; running && !terminating {
+		ctx = logging.NewContext(ctx, logging.FromContext(ctx).WithValues("pod", pod.Name))
+
+		podExecutor = func(
+			_ context.Context, stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			return r.PodExec(pod.Namespace, pod.Name, container, stdin, stdout, stderr, command...)
+		}
+	}
+	if podExecutor == nil {
+		return nil
+	}
+
+	// Get pgAdmin version. If it is not v8 or higher, return early
+	// as user management is only supported for pgAdmin v8 and higher.
+	pgadminVersion, err := r.reconcilePGAdminMajorVersion(ctx, podExecutor)
+	if err != nil {
+		return err
+	}
+	if pgadminVersion < 8 {
+		// If pgAdmin version is less than v8 and user management is being attempted,
+		// log a message clarifying that it is only supported for pgAdmin v8 and higher.
+		if len(pgadmin.Spec.Users) > 0 {
+			log.Info(fmt.Sprintf(
+				"The pgAdmin major version is v%v. User management is only supported for pgAdmin v8 and higher.",
+				pgadminVersion,
+			))
+		}
+		return err
+	}
+
+	return r.writePGAdminUsers(ctx, pgadmin, podExecutor)
+}
+
+// reconcilePGAdminMajorVersion execs into the pgAdmin pod and retrieves the pgAdmin major version
+func (r *PGAdminReconciler) reconcilePGAdminMajorVersion(ctx context.Context, exec Executor) (int, error) {
+	script := `
+PGADMIN_DIR=/usr/local/lib/python3.11/site-packages/pgadmin4
+cd $PGADMIN_DIR && python3 -c "import config; print(config.APP_RELEASE)"
+`
+
+	var err error
+	var stdin, stdout, stderr bytes.Buffer
+
+	err = exec(ctx, &stdin, &stdout, &stderr,
```

Comment:
Very much :pick:, but
```suggestion
	err := exec(ctx, &stdin, &stdout, &stderr,
```
and remove `var err error` above.